scripts/openstack-update.sh

```diff
@@ -115,8 +115,24 @@ oc wait $OPENSTACK_VERSION_CR --for=condition=MinorUpdateOVNControlplane --timeo
 get_current_state "02_after_ovn_controlplane_update"
 
 # start ovn update on data plane
-DATAPLANE_NODESET=$(oc get openstackdataplanenodeset -o name | awk -F'/' '{print "    - "  $2}')
+nodes_with_ovn=()
+# Get the names of all OpenStackDataPlaneNodeSet resources
+node_names=$(oc get openstackdataplanenodeset -o name | awk -F'/' '{print $2}')
+
+# Loop through each OpenStackDataPlaneNodeSet
+while IFS= read -r node_name; do
```

Comment:
As per my previous comment this could then be:

```bash
while read -r node_name services; do
    echo $node_name;
    for item in ${services[@]};do
        [[ "ovn" =~ $item ]] && nodes_with_ovn+=("- $node_name")
    done
done <<<$EDPM
```

Reply:
This is not working without split of node_name string anyway.
The structure of output from previous command is `openstack-edpm   [bootstrap download-cache configure-network validate-network install-os configure-os ssh-known-hosts run-os reboot-os install-certs ovn neutron-metadata libvirt nova]`. It is single line string, and for `item in ${services[@]};do` to work the string has to be  split and cast to array.

Reply:
if you `read -r node_name services;` you're splitting them using the `read` builtin, and service is not a single string.

```bash
 EDPM=$(oc get openstackdataplanenodeset -o custom-columns=NAME:.metadata.name,SERVICES:.spec.services --no-headers)
while read -r node_name services; do
    echo $node_name; 
    for item in ${services[@]};do
        echo $item;
    done
done <<<$EDPM
```

and this is the output I get:

```bash
ceph-nodes
[bootstrap
configure-network
validate-network
install-os
ceph-hci-pre]
openstack-edpm
[bootstrap
configure-network
validate-network
install-os
configure-os
ssh-known-hosts
run-os
reboot-os
install-certs
ceph-client
ovn
neutron-metadata
libvirt
nova-custom-ceph]
```

Reply:
Thank you @fmount. I had `while read IF`. Once I fixed that it is working as explained in your snippet.